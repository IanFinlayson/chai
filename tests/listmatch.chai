
def doJunk(coords [Int]):
    match coords:
        case []:
            print("0 dimensional")
        case [0]:
            print("1 dimensional, but zero :(")
        case [x]:
            print("1 dimensional")
        case [x, y]:
            print("2 dimensional")
        case [x, y, z]:
            print("3 dimensional!!")
        case ls:
            print("MORE dimensions??")

def main():
    doJunk([])
    doJunk([4])
    doJunk([0])
    doJunk([7, 3])
    doJunk([88, 2, 34])
    doJunk([9, 1, 3, 5])

