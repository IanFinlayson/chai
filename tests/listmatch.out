0 dimensional
1 dimensional
1 dimensional, but zero :(
2 dimensional
3 dimensional!!
MORE dimensions??
